### PR TITLE
FOUR-21263 implement a general method to index cache keys

### PR DESCRIPTION
**Pull request by devmiguelangel**
## Issue & Reproduction Steps
As a developer, I would like to have a method to index and find keys on the Cache implementation.

## Solution
- Add a new abstract class `CacheManagerBase`
- Implement the `getKeysByPattern` method
- Add a console command to clear settings cache

## How to Test
Run `php artisan test --filter='SettingCacheTest|CacheManagerBaseTest'`

## Related Tickets & Packages
[FOUR-21263](https://processmaker.atlassian.net/browse/FOUR-21263)

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

[FOUR-21263]: https://processmaker.atlassian.net/browse/FOUR-21263?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/Feature/Cache/CacheManagerBaseTest.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        Redis::shouldReceive('connection')
+            ->with('redis')
+            ->andReturnSelf();
+
+        Redis::shouldReceive('keys')
+            ->with($prefix . '*')
+            ->andThrow(new Exception('Redis error'));
+
+        Log::shouldReceive('info')
+            ->with('CacheABC' . 'Redis error')
```
Comment (caleeli) [inline tests/Feature/Cache/CacheManagerBaseTest.php:100]: Update this label related to the class CacheManagerBase

File: tests/Feature/Cache/SettingCacheTest.php [hunk 2]
```diff
@@ -210,7 +215,7 @@ public function testTryClearByPatternWithNonRedisDriver()
     {
         config()->set('cache.default', 'array');
 
-        $this->expectException(SettingCacheException::class);
+        $this->expectException(\Exception::class);
```
Comment (caleeli) [inline tests/Feature/Cache/SettingCacheTest.php:218]: Why this was changed to a generic Exception, seems it should return SettingCacheException

File: tests/Feature/Cache/CacheManagerBaseTest.php [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        $this->assertCount(0, $result);
+    }
+
+    public function testGetKeysByPatternWithInvalidConnection()
+    {
+        config()->set('cache.default', 'array');
+
+        $this->cacheManagerBase = $this->getMockForAbstractClass(CacheManagerBase::class);
+
+        $this->expectException(Exception::class);
```
Comment (caleeli) [inline tests/Feature/Cache/CacheManagerBaseTest.php:78]: Please create a specific Exception class

File: ProcessMaker/Cache/CacheManagerBase.php [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+     * Retrieve an array of cache keys that match a specific pattern.
+     *
+     * @param string $pattern The pattern to match.
+     *
+     * @return array An array of cache keys that match the pattern.
+     */
+    public function getKeysByPattern(string $pattern): array
+    {
+        if (!in_array($this->connection, self::AVAILABLE_CONNECTIONS)) {
+            throw new Exception('`getKeysByPattern` method only supports Redis connections.');
```
Comment (caleeli) [inline ProcessMaker/Cache/CacheManagerBase.php:48]: Please create a specific Exception class

**Review by caleeli — CHANGES_REQUESTED** [review]
Please review the comments

File: ProcessMaker/Cache/CacheManagerBase.php (hunk 4 above)
Comment (devmiguelangel) [reply ProcessMaker/Cache/CacheManagerBase.php]: A specific exception class was added

File: tests/Feature/Cache/CacheManagerBaseTest.php (hunk 3 above)
Comment (devmiguelangel) [reply tests/Feature/Cache/CacheManagerBaseTest.php]: A specific exception class was added

File: tests/Feature/Cache/CacheManagerBaseTest.php (hunk 1 above)
Comment (devmiguelangel) [reply tests/Feature/Cache/CacheManagerBaseTest.php]: The Log label was updated

File: tests/Feature/Cache/SettingCacheTest.php (hunk 2 above)
Comment (devmiguelangel) [reply tests/Feature/Cache/SettingCacheTest.php]: My mistake, it was fixed

**Review by caleeli — APPROVED** [review]

**Pull request merged**